ios/example/app.js

```diff
@@ -1,90 +1,176 @@
-    });
-    mifare.connect({mifare});
-    nfcAdapter.invalidate(sessionType);
+    if (e.errorCode){
+        Ti.API.info('Error when detecting tag with - error code ' + e.errorCode + ' error domain: ' + e.errorDomain + ' error description ' + e.errorDescription);
+        logs.push('Error when detecting tag with - error code ' + e.errorCode + ' error domain: ' + e.errorDomain + ' error description ' + e.errorDescription);
+        setData(logs);
+    }else{
+        logs.push('Tag scanning start');
+        logs.push('Mifare tag detected');
+        setData(logs);
+    }
+  var mifare = nfcAdapter.createTagTechMifareUltralight({'tag':e.tags[0]});
+  mifare.addEventListener('didConnectTag', function (e) {
+      if (e.errorCode){
+          Ti.API.info('Error when connectiong tag with - error code ' + e.errorCode + ' error domain: ' + e.errorDomain + ' error description ' + e.errorDescription);
+          logs.push('Error when connectiong tag with - error code ' + e.errorCode + ' error domain: ' + e.errorDomain + ' error description ' + e.errorDescription);
+          setData(logs);
+      }else{
+          Ti.API.info('Connected tag object : ' + e.tag);
+          logs.push('Mifare tag connected : '  + e.tag);
+          setData(logs);
+      }
+      var sendMiFareCommand = mifare.sendMiFareCommand({data: [0x30,0x04]})
+      mifare.addEventListener('didSendMiFareCommand', function (e) {
+          if (e.errorCode){
+              Ti.API.info('Error when sending command with - error code ' + e.errorCode + ' error domain: ' + e.errorDomain + ' error description ' + e.errorDescription);
+              logs.push('Error when sending command with - error code ' + e.errorCode + ' error domain: ' + e.errorDomain + ' error description ' + e.errorDescription);
+              setData(logs);
+          }else {
+              Ti.API.info('Connected tag object : ' + e.responseDataLength);
+              logs.push('Command send to Mifare tag to read page data');
+              logs.push('Length of response object : ' + e.responseDataLength);
+              setData(logs);
+          }
+      });
+      nfcAdapter.invalidate(sessionType);
+  });
+   if (mifare){
+  mifare.connect({mifare});
+   }else {
+       alert('No mifare tag available to connect');
+   }
 });
 nfcAdapter.addEventListener('didInvalidateWithError', function (e) {
-        Ti.API.info('code: ' + e.code);
+  Ti.API.info('Invalidate the session with - error code ' + e.errorCode + ' error domain: ' + e.errorDomain + ' error description ' + e.errorDescription);
+  logs.push('Invalidate the session with - error code ' + e.errorCode + ' error domain: ' + e.errorDomain + ' error description ' + e.errorDescription);
+  setData(logs);
+});
+var clearLogBtn = Titanium.UI.createButton({
+  left: 50,
+  top: 150,
+  title: 'Clear Log'
+});
+
+clearLogBtn.addEventListener('click', function() {
+   var rowCount = logs.length;
+   for (var i = rowCount ; i > -1; i--) {
```

Comment:
@anil-shukla-axway, i do no think any loop is required to clear the logs. You can clear logs like...

```suggestion
   logs.splice(0, logs.length);
    setData(logs);
```

Reply:
Updated